templates/buildkite-elastic.yml

```diff
@@ -598,8 +586,47 @@ Resources:
   AgentAutoScaleGroup:
     Type: AWS::AutoScaling::AutoScalingGroup
     Properties:
-      VPCZoneIdentifier:
-        { "Fn::If": [ "CreateVpcResources", [ { Ref: Subnet0 }, { Ref: Subnet1 } ], { Ref: Subnets } ] }
+      VPCZoneIdentifier: !If [
+        CreateVpcResources,
+        !If [
```

Comment:
This can be simplified to:

```
!If
  - CreateVpcResources
  - !Split
    - ','
    # This eliminates `AWS::NoValue` entries
    - !Join 
      - ','
      - !Ref PrivateSubnet1
      - !Ref PrivateSubnet2
      - !Ref PublicSubnet1
      - !Ref PublicSubnet2
      - !If [HasPrivateSubnet3, !Ref PrivateSubnet3, !Ref 'AWS::NoValue']
      - !If [HasPrivateSubnet4, !Ref PrivateSubnet4, !Ref 'AWS::NoValue']
      - !If [HasPrivateSubnet5, !Ref PrivateSubnet5, !Ref 'AWS::NoValue']
      - !If [HasPrivateSubnet6, !Ref PrivateSubnet6, !Ref 'AWS::NoValue']
      - !If [HasPublicSubnet3, !Ref PublicSubnet3, !Ref 'AWS::NoValue']
      - !If [HasPublicSubnet4, !Ref PublicSubnet4, !Ref 'AWS::NoValue']
      - !If [HasPublicSubnet5, !Ref PublicSubnet5, !Ref 'AWS::NoValue']
      - !If [HasPublicSubnet6, !Ref PublicSubnet6, !Ref 'AWS::NoValue']
```